plutus-conformance/agda/Spec.hs

```diff
@@ -0,0 +1,42 @@
+{- | Conformance tests for the Agda implementation. -}
+
+module Main (main) where
+
+import Control.Monad.Error.Class
+import Control.Monad.Trans.Except
+import MAlonzo.Code.Main (runUAgda)
+import PlutusConformance.Common
+import PlutusCore (Error (..))
+import PlutusCore.Default
+import PlutusCore.Quote
+import UntypedPlutusCore qualified as UPLC
+import UntypedPlutusCore.DeBruijn
+
+-- | Our `evaluator` for the Agda UPLC tests is the CEK machine.
+agdaEvalUplcProg :: UplcProg -> Maybe UplcProg
+agdaEvalUplcProg p =
+    let
+        -- turn a UPLC program to a UPLC term
+        tmU = UPLC.mkTerm p
+        -- turn it into an untyped de Bruijn term
+        tmUDB :: ExceptT FreeVariableError Quote (UPLC.Term NamedDeBruijn DefaultUni DefaultFun ())
+        tmUDB = deBruijnTerm tmU
+    in
+    case runQuote $ runExceptT $ withExceptT FreeVariableErrorE tmUDB of
+        -- if there's an exception, evaluation failed, should return `Nothing`.
+        Left _ -> Nothing
```

Comment:
But it's not evaluation, it's conversion to De Bruijn indices that failed? Or am I misreading it?

Reply:
Again, this is the open-term check, effectively.